base/reinterpretarray.jl

```diff
@@ -376,12 +376,10 @@ elsize(::Type{<:ReinterpretArray{T}}) where {T} = sizeof(T)
 cconvert(::Type{Ptr{T}}, a::ReinterpretArray{T,N,S} where N) where {T,S} = cconvert(Ptr{S}, a.parent)
 unsafe_convert(::Type{Ptr{T}}, a::ReinterpretArray{T,N,S} where N) where {T,S} = Ptr{T}(unsafe_convert(Ptr{S},a.parent))
 
-@propagate_inbounds function getindex(a::NonReshapedReinterpretArray{T,0,S}) where {T,S}
-    if isprimitivetype(T) && isprimitivetype(S)
-        reinterpret(T, a.parent[])
-    else
-        a[firstindex(a)]
-    end
+@propagate_inbounds function getindex(a::ReinterpretArray{T,0,S,IsReshaped}) where {T,S,IsReshaped}
+    check_readable(a)
+    check_ptr_indexable(a) && return _getindex_ptr(a)
+    return reinterpret(T, a.parent[])
```

Comment:
L382 is valid only for `NonReshapedReinterpretArray`, it would break `reinterpret(reshape, Tuple{Int8,Int}, view([1,1], [1,2]))`.